src/lib/y2storage/proposal/autoinst_devices_planner.rb

```diff
@@ -418,6 +418,8 @@ def remove_shadowed_subvols(planned_devices)
           subvols_added =
             device.respond_to?(:mount_point) && device.mount_point == "/" && @default_subvolumes_used
 
+          # Use #dup to avoid bsc#1084213
```

Comment:
I would `dup` the list just when it is assigned from the settings:
https://github.com/ancorgs/yast-storage-ng/blob/b59dafb1b2e7c1d58d35e914907ea7fe7b0ef7dc/src/lib/y2storage/proposal/autoinst_devices_planner.rb#L233

This would avoid possible similar errors in other places.

Reply:
There are at least three assignations of that. So duping all of them looks wrong.

At most, I would define a setter. Like
```ruby
module Planned::CanBeFormatted
  def subvolumes=(value)
    @subvolumes = value.dup
  end
end
```

That was indeed my first option, but I somehow though it would have been more surprising. Less rubyist, sort to say.

I don't care much, either solution (the one on this PR or redefining the setter) are acceptable to me.

Reply:
So:

- Solution 1: the one currently on this PR. It's correct but @joseivanlopez pointed that we can make the same error in the future in another part.
- Solution 2: redefining the setter. It's also correct and doesn't have the mentioned problem, although it has slightly surprising implications (from the OOP rubyist POV).
- Solution 3: a `dup` on each assignation as suggested by @joseivanlopez. I don't like it. Very error prone.

I discussed offline with @imobachgs and he would go for 2. Since 2 is also pretty aligned with @joseivanlopez's concerns, I will switch from 1 to 2.